src/lib/format.js

```diff
@@ -0,0 +1,115 @@
+    return string.charAt(0).toUpperCase() + string.slice(1);
+};
+
+export const pascalToCamelCase = str => {
+    return str.split('_').map((word, index) => {
+        // If it is the first word make sure to lowercase all the chars.
+        if (index === 0) {
+            return word.toLowerCase();
+        }
+        // If it is not the first word only upper case the first char and lowercase the rest.
+        return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
+    }).join('');
+};
+
+// turn humpdey_dumpdey into Humpdey Dumpdey
+export const humanize = (str) => {
+    const frags = str.split('_').join(', ').split('-').join(', ').split(', ');
+    for (let i = 0; i < frags.length; i++) {
+        frags[i] = capitalizeFirstLetter(frags[i].toLowerCase());
+    }
+    return frags.join(' ');
+};
+
+/**
+ * Pretty print numbers
+ * @param number
+ * @return {string}
+ */
+export const digitGroupSeparator = number => {
+    if (number && typeof number === 'number') {
+        return number.toLocaleString();
+    }
+    return '0';
+};
+
+/**
+ * Round the number to the precision specified
+ */
+export const formatNumber = (number, precision) => Number.parseFloat(number).toFixed(precision);
+
+export const pluralize = (word, num) => (num === 1 ? word : `${word}s`);
+
+/**
+ * Return the optimal number and the size label for data in bytes
+ * https://stackoverflow.com/questions/15900485/correct-way-to-convert-size-in-bytes-to-kb-mb-gb-in-javascript
+ */
+export const formatBytes = (bytes, decimals = 2) => {
+    if (bytes === 0 || bytes === undefined) return { value: 0, size: 'B' };
+
+    const k = 1024;
+    const dm = decimals < 0 ? 0 : decimals;
+    const sizes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+
+    let i = Math.floor(Math.log(bytes) / Math.log(k));
+
+    let value = (bytes / Math.pow(k, i)).toFixed(dm);
+    let size = sizes[i];
+
+    if (value === k.toFixed(dm)) {
+        i = i + 1;
+
+        value = (bytes / Math.pow(k, i)).toFixed(dm);
+        size = sizes[i];
+    }
+
+    return { value: value, size: size };
+};
+
+/**
+ * Formats Latitude and Longitude numbers to a uniform decimal place without rounding on truncation
+ * @param {number} value - The lat/long value to format
+ * @returns {string}
+ */
+export const formatLatLong = (value) => {
+    if (!value) return '';
+    const [integerPart, fractionalPart] = value.toString().split('.');
+
+    return `${integerPart}.${fractionalPart.substring(0, 5)}`;
+};
+
+/**
+ * Receives a celcius temperature and returns a properly formatted temperature string
+ * with degree symbol. An optional measurementSystem param can be provided to output
+ * as Fahrenheit.
+ * @param {number} celciusTemp - The temperature in celcius
+ * @param {string} [measurementSystem="metric"] - The output measurement system
+ * @returns {string}
+ */
+export const formatCelciusTemperature = (celciusTemp = 0, measurementSystem = 'metric') => {
```

Comment:
This should be spelled "celsius".  Alternatively, we could just say "metric" (in contrast to "imperial").

Reply:
No sure if I understood this. Alternatively, we could just say "metric" (in contrast to "imperial"). I have corrected the spelling though.